sklearn/semi_supervised/_label_propagation.py

```diff
@@ -127,6 +129,17 @@ def _get_kernel(self, X, y=None):
                 return rbf_kernel(X, X, gamma=self.gamma)
             else:
                 return rbf_kernel(X, y, gamma=self.gamma)
+        elif self.kernel == "sparse-rbf":
+            self.nn_fit = NearestNeighbors(self.n_neighbors,
+                                           n_jobs=self.n_jobs).fit(X)
+            # rbf(x1, x2) = exp(-gamma * ||x1 - x2||^2)
+            W = self.nn_fit.kneighbors_graph(y, mode='distance').T.power(2)
+            W *= -1 * self.gamma
+            np.exp(W.data, out=W.data)
+            # explicitly set diagonal,
+            # since np.exp(W.data) does not modify zeros on the diagonal
+            W.setdiag(1)
```

Comment:
Note that this line causes a warning about changing the sparsity structure.

I think the 2 best options here are:
1. Suppress warning (if starting with `lil` matrix and converting to `csc` is actually slower overall)
2. Start with `lil` matrix and convert to `csc` after. 

My guess is that the correct thing to do is suppress the warning, but I left it for now so others could weigh in on this.